gokart/utils.py

```diff
@@ -71,6 +71,19 @@ def flatten(targets: FlattenableItems[T]) -> list[T]:
     return flat
 
 
+K = TypeVar('K')
+
+
+def map_flattenable_items(items: FlattenableItems[T], func: Callable[[T], K]) -> FlattenableItems[K]:
+    if isinstance(items, dict):
+        return {k: map_flattenable_items(v, func) for k, v in items.items()}
+    if isinstance(items, str):
+        return items  # type: ignore
+    if isinstance(items, Iterable):
+        return [map_flattenable_items(i, func) for i in items]
```

Comment:
When pass `tuple[T]`, it should returns `tuple[K]`. But, in this implementation, this case is not cared for.

Reply:
And, could you add testcase?